### PR TITLE
fix  clean up participant before deleting.

### PR DESCRIPTION
**Pull request by tfoote**
Fix cleanup on node destruction. 

connects to ros2/system_tests#10

fixes the ticket  ros2/system_tests#29

### CODE REVIEW

File: rmw_opensplice_cpp/src/functions.cpp [hunk 1]
```diff
@@ -419,6 +419,11 @@ rmw_destroy_node(rmw_node_t * node)
     RMW_SET_ERROR_MSG("participant handle is null");
     return RMW_RET_ERROR;
   }
+
+  // This unregisters types and destroys topics which were shared between
+  // publishers and subscribers and could not be cleaned up in the delete functions.
+  participant->delete_contained_entities();
```
Comment (dirk-thomas) [inline rmw_opensplice_cpp/src/functions.cpp]: Please handle the return value.

**Comment by tfoote** [comment]
updated to catch return code

**Comment by esteve** [comment]
+1

**Comment by dirk-thomas** [comment]
+1

Should this also be done for the rmw_connext implementations? Or is that done automatically for Connext but now for OpenSplice?

**Comment by tfoote** [comment]
The original version of this PR without the error handling is the state in rmw_connext.

**Pull request merged**